pyomo/core/tests/unit/test_derivs.py

```diff
@@ -172,6 +172,21 @@ def test_atan(self):
         self.assertAlmostEqual(derivs[m.x], pyo.value(symbolic[m.x]), tol+3)
         self.assertAlmostEqual(derivs[m.x], approx_deriv(e, m.x), tol)
 
+    def test_abs(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var(initialize=2.0)
+        e = 2 * abs(m.x)
+        with self.assertRaises(DifferentiationException):
```

Comment:
In general, I prefer `assertRaisesRegex()` so that we check that the correct message is being raised (I have run across bugs where we thought we were testing that Pyomo correctly handled an exception only to find that there was a bug that happened to raise the same exception).

Reply:
Good point. Fixed.